pkg/server/container_create_unix.go

```diff
@@ -168,10 +168,7 @@ func (c *criService) containerSpec(id string, sandboxID string, sandboxPid uint3
 		if !sandboxConfig.GetLinux().GetSecurityContext().GetPrivileged() {
 			return nil, errors.New("no privileged container allowed in sandbox")
 		}
-		specOpts = append(specOpts, oci.WithPrivileged)
-		if !ociRuntime.PrivilegedWithoutHostDevices {
```

Comment:
could we add some comments here... I don't follow why are we removing/ignoring the privilegedwithouthostdevices config directive and replacing it with always withhostdevices when privileged.  

	// PrivilegedWithoutHostDevices overloads the default behaviour for adding host devices to the
	// runtime spec when the container is privileged. Defaults to false.